mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMvRxViewModel.kt

```diff
@@ -70,8 +71,10 @@ abstract class BaseMvRxViewModel<S : MvRxState>(
         initialState::class.declaredMemberProperties.asSequence()
             .filter { it.visibility == KVisibility.PUBLIC }
             .forEach { prop ->
-                @Suppress("UNCHECKED_CAST")
-                (prop as? KProperty1<S, Any?>)?.get(initialState)
+                if (prop.findAnnotation<PersistState>() != null) {
```

Comment:
@tasomaniac Does this actually improve performance? Finding annotations might be as slow as just accessing the field.

Reply:
This is my naive approach. I didn't know that findAnnotation would be expensive. In our project we don't use PersistState at all so I was wondering how can we avoid warming up reflection since we don't really need it.

Reply:
@tasomaniac You could measure it by sticking in some logs in a class that has a lot of both PersistState and not PersistState properties.

Reply:
So I tried to do that but couldn't find a noticable difference.

While doing that I also realized that we load all annotations for all fields. So finding the annotation is instant. 

It's up to you to merge this. I don't think it would make it worse.